sgkit/tests/test_window.py

```diff
@@ -0,0 +1,46 @@
+import allel
+import dask.array as da
+import numpy as np
+import pytest
+
+from sgkit.window import moving_statistic
+
+
+@pytest.mark.parametrize(
+    "length, chunks, size, step", [(12, 6, 4, 4), (12, 6, 4, 2), (12, 5, 4, 4)]
+)
+def test_moving_statistic_1d(length, chunks, size, step):
+    values = da.from_array(np.arange(length), chunks=chunks)
+
+    stat = moving_statistic(values, np.sum, size=size, step=step, dtype=values.dtype)
+    stat = stat.compute()
+    if length % size != 0 or size != step:
+        # scikit-allel misses final window in this case
+        stat = stat[:-1]
+
+    values_sa = np.arange(length)
+    stat_sa = allel.moving_statistic(values_sa, np.sum, size=size, step=step)
+
+    np.testing.assert_equal(stat, stat_sa)
+
+
+@pytest.mark.parametrize(
+    "length, chunks, size, step", [(12, 6, 4, 4), (12, 6, 4, 2), (12, 5, 4, 4)]
+)
+def test_moving_statistic_2d(length, chunks, size, step):
+    arr = np.arange(length * 3).reshape(length, 3)
+
+    def sum_cols(x):
+        return np.sum(x, axis=0)
+
+    values = da.from_array(arr, chunks=chunks)
+    stat = moving_statistic(values, sum_cols, size=size, step=step, dtype=values.dtype)
+    stat = stat.compute()
+    if length % size != 0 or size != step:
+        # scikit-allel misses final window in this case
+        stat = stat[:-1]
+
+    values_sa = arr
+    stat_sa = allel.moving_statistic(values_sa, sum_cols, size=size, step=step)
+
+    np.testing.assert_equal(stat, stat_sa)
```

Comment:
Would it make sense to add more tests here, there are some interesting functions in the `window` package, like `_get_chunked_windows`? Plus test for corner cases like empty arrays etc. And maybe also a test for what happens when you window multiple times with `merge=True`. And maybe a test case for https://github.com/pystatgen/sgkit/pull/303#issuecomment-705682385

Reply:
Added tests for various window functions in https://github.com/pystatgen/sgkit/pull/303/commits/d57aefdb889fd9190b44d4439ba75e91ac8201b9